File: js/1047.js
var removeDuplicates = function(S) {
    var i = -1;
    var j = 0;
    var array = S.split("");
    while (j < array.length) {
        if (i === -1) {
            array[++i] = array[j++];
            continue;
        }
        if (array[i] == array[j]) {
            i--;
            j++;
        } else {
            i++;
            array[i] = array[j];
            j++;
        }
    }
    return array.slice(0,i+1).join("");
};

removeDuplicates("abbaca");

File: c/1047.c
char * removeDuplicates(char * S){
    int i = -1;
    int j = 0;
    while (S[j] != '\0') {
        if (i == -1) {
            S[++i] = S[j++];
            continue;
        }
        if (S[i] == S[j]) {
            i--;
            j++;
        } else {
            S[++i] = S[j++];
        }
        
    }
    while (S[++i] != '\0'){
        S[i] = '\0';
    }
    return S;
}
